flang/test/Lower/real.f90

```diff
@@ -0,0 +1,104 @@
+! RUN: bbc %s -o - | FileCheck %s
+
+! Test real add on real kinds.
+
+! CHECK-LABEL: real2
+REAL(2) FUNCTION real2(x0, x1)
+  REAL(2) :: x0
+  REAL(2) :: x1
+  ! CHECK-DAG: %[[v1:.+]] = fir.load %arg0 : !fir.ref<f16>
+  ! CHECK-DAG: %[[v2:.+]] = fir.load %arg1 : !fir.ref<f16>
+  ! CHECK: %[[v3:.+]] = fir.addf %[[v1]], %[[v2]] : f16
+  real2 = x0 + x1
+  ! CHECK: return %{{.*}} : f16
+END FUNCTION real2
+
+! CHECK-LABEL: real3
+REAL(3) FUNCTION real3(x0, x1)
+  REAL(3) :: x0
+  REAL(3) :: x1
+  ! CHECK-DAG: %[[v1:.+]] = fir.load %arg0 : !fir.ref<bf16>
+  ! CHECK-DAG: %[[v2:.+]] = fir.load %arg1 : !fir.ref<bf16>
+  ! CHECK: %[[v3:.+]] = fir.addf %[[v1]], %[[v2]] : bf16
+  real3 = x0 + x1
+  ! CHECK: return %{{.*}} : bf16
+END FUNCTION real3
+
+! CHECK-LABEL: real4
+REAL(4) FUNCTION real4(x0, x1)
+  REAL(4) :: x0
+  REAL(4) :: x1
+  ! CHECK-DAG: %[[v1:.+]] = fir.load %arg0 : !fir.ref<f32>
+  ! CHECK-DAG: %[[v2:.+]] = fir.load %arg1 : !fir.ref<f32>
+  ! CHECK: %[[v3:.+]] = fir.addf %[[v1]], %[[v2]] : f32
+  real4 = x0 + x1
+  ! CHECK: return %{{.*}} : f32
+END FUNCTION real4
+
+! CHECK-LABEL: defreal
+REAL FUNCTION defreal(x0, x1)
```

Comment:
Isn't the default REAL choice subject to command line option values?

Reply:
Did we add such options to bbc?

Reply:
Whatever the current driver status is, default type options are common across compilers.  nvfortran has _-M[no]r8_, _-M[no]r8intrinsics[=float]_,  _-r4_, _-r8_.

Reply:
I understand your point. This test is specifically written for `bbc` though.